src/cosim_toolbox/cosim_toolbox/federate.py

```diff
@@ -69,140 +70,96 @@ class for Python-based federates in HELICS. It outlines the typical
         federation (dict): Dictionary with all federate configuration information
     """
 
-    def __init__(self, fed_name="", use_mdb=True, use_pdb=True, **kwargs):
-        self.hfed: h.HelicsFederate = None
-        self.mddb: DBConfigs = None
-        self.config: dict = None
-        self.scenario: dict = None
-        self.scenario_name: str = None
-        self.federation: dict = None
-        self.federation_name: str = None
-        self.federate: dict = None
-        self.federate_type: str = None
+    def __init__(
+        self,
+        fed_name: str = "",
+        use_mdb: bool = True,
+        use_pdb: bool = True,
+        metadata_config: Optional[Dict[str, Any]] = None,
+        timeseries_config: Optional[Dict[str, Any]] = None,
+        metadata_location: Optional[str] = None,
+        timeseries_location: Optional[str] = None,
+        *,
+        debug: bool = True,
+    ):
+        """Initializes the Federate.
+
+        Args:
+            fed_name (str, optional): The name of the federate. Defaults to "".
+            use_mdb (bool, optional): Whether to use a metadata database (e.g., MongoDB). Defaults to True.
```

Comment:
Is this choosing between MongoDB and JSON files? If so, the description needs to be updated to something more direct: "Set to `True` to use MongoDB for metadata backend, `False` for JSON files on local disk".

Similarly with `use_pdb`.